src/meta/IndexManager.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef META_INDEXMANAGER_H_
+#define META_INDEXMANAGER_H_
+
+#include "base/Base.h"
+#include "meta/client/MetaClient.h"
+#include "interface/gen-cpp2/meta_types.h"
+
+namespace nebula {
+namespace meta {
+
+class IndexManager {
+public:
+    ~IndexManager() = default;
+
+    static std::unique_ptr<IndexManager> create();
+
+    virtual void init(MetaClient *client) = 0;
+
+    virtual StatusOr<std::shared_ptr<nebula::cpp2::IndexItem>>
+    getTagIndex(GraphSpaceID space, IndexID index) = 0;
+
+    virtual StatusOr<std::shared_ptr<nebula::cpp2::IndexItem>>
+    getEdgeIndex(GraphSpaceID space, IndexID index) = 0;
+
+    virtual StatusOr<std::vector<std::shared_ptr<nebula::cpp2::IndexItem>>>
+    getTagIndexes(GraphSpaceID space) = 0;
```

Comment:
Maybe listTagIndexes?

Reply:
This is because we should avoid distinguish show tag index